src/python/pants/core/util_rules/adhoc_process_support.py

```diff
@@ -266,14 +277,16 @@ async def _resolve_runnable_dependencies(
     shim_digest_path = f"_runnable_dependency_shims_{shim_digest.fingerprint}"
     immutable_input_digests = {shim_digest_path: shim_digest}
     _safe_update(immutable_input_digests, merged_extras.immutable_input_digests)
+    environment = {"_PANTS_SHIM_ROOT": "{chroot}"}
+    environment.update(runnable_extra_env_vars)
```

Comment:
We should switch the update order here, we never want to stomp _PANTS_SHIM_ROOT (although it's highly unlikely)

Reply:
Reversed the order